### PR TITLE
kernel/mqueue: Fix an error when a task with open message queue descriptors is killed via task_delete().

### PR DESCRIPTION
**Pull request by jeongarmy**
Fix an error when a task with open message queue descriptors is killed via task_delete().  Noted by Anton Gropyanov.

Conflicts:
	fs/mqueue/mq_close.c
	fs/mqueue/mq_open.c
	fs/mqueue/mq_unlink.c
	include/nuttx/mqueue.h
	sched/mqueue/mq_desclose.c
	sched/mqueue/mq_descreate.c
	sched/mqueue/mq_release.c
	sched/mqueue/mqueue.h
	sched/task/task_exithook.c

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [e118336909f4db31621599dc9ca6c667dd034e54] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e118336909f4db31621599dc9ca6c667dd034e54] - Code Rule Check OK.

File: os/fs/mqueue/mq_close.c [hunk 1]
```diff
@@ -155,6 +129,54 @@ int mq_close(mqd_t mqdes)
...
+	FAR struct tcb_s *rtcb = (FAR struct tcb_s *)sched_self();
+	int ret;
+
+	/* Lock the scheduler to prevent any asynchrounous task delete operation
+	* (unlikely).
+	*/
+
+	sched_lock();
+
+	rtcb = (FAR struct tcb_s *)sched_self();
```
Comment (sunghan-chang) [inline os/fs/mqueue/mq_close.c]: duplicated with line 158. Let's remove this with another commit.

**Comment by seinfra** [comment]
Target : [a77744e590f73d2a7df05d27f53b1821ea5013ec] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a77744e590f73d2a7df05d27f53b1821ea5013ec] - Code Rule Check OK.

File: os/fs/mqueue/mq_close.c [hunk 2]
```diff
@@ -155,7 +155,7 @@ int mq_close_group(mqd_t mqdes, FAR struct task_group_s *group)
 
 int mq_close(mqd_t mqdes)
 {
-	FAR struct tcb_s *rtcb = (FAR struct tcb_s *)sched_self();
+	FAR struct tcb_s *rtcb;
```
Comment (sunghan-chang) [inline os/fs/mqueue/mq_close.c]: We don't need to get this after sched_lock. Let's remove line 167 instead of deleting here.

**Comment by seinfra** [comment]
Target : [cb701c6d5e50c84e8a25e425081ab1f759220e5d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cb701c6d5e50c84e8a25e425081ab1f759220e5d] - Code Rule Check OK.

File: os/fs/mqueue/mq_close.c (hunk 2 above)
Comment (sunghan-chang) [reply os/fs/mqueue/mq_close.c]: @jeongarmy You didn't follow my suggestion. Please let me know your opinion.

**Comment by seinfra** [comment]
Target : [b59385040e2aec9afcafd84c26cff5dc54d0751c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b59385040e2aec9afcafd84c26cff5dc54d0751c] - Code Rule Check OK.

**Pull request merged**